src/backend/distributed/metadata/metadata_cache.c

```diff
@@ -402,19 +421,49 @@ IsCitusTable(Oid relationId)
 
 
 /*
- * IsCitusTableViaCatalog returns whether the given relation is a
- * distributed table or not.
- *
- * It does so by searching pg_dist_partition, explicitly bypassing caches,
- * because this function is designed to be used in cases where accessing
- * metadata tables is not safe.
- *
- * NB: Currently this still hardcodes pg_dist_partition logicalrelid column
- * offset and the corresponding index.  If we ever come close to changing
- * that, we'll have to work a bit harder.
+ * IsCitusTableTypeViaCatalog returns true if the given table entry belongs to
+ * the given table type group by performing index-scan on pg_dist_partition.
+ */
+static bool
+IsCitusTableTypeViaCatalog(Oid relationId, CitusTableType tableType)
+{
+	char partitionMethod = 0;
+	char replicationModel = REPLICATION_MODEL_INVALID;
+	bool isCitusTable = IsCitusTableViaCatalogInternal(relationId, &partitionMethod,
```

Comment:
For example one idea would be to return a cache entry from this method with the necessary fields filled and we can just check if it is `NULL` for `isCitusTable` which would simplify the API of  `IsCitusTableViaCatalogInternal`, it would just take the relation id.

Reply:
(also replying https://github.com/citusdata/citus/pull/4188#discussion_r492075602)

Building a mock cache entry for the existing API was also the initial idea that came to my mind but I think it might make the things seem to be more complicated than it is.

Now we still have only one method to change and we can increase the number of parameters that it requires in future if we want I think. What do you think ?